v2/app/src/main/java/com/google/firebase/example/makeitso/data/datasource/AuthRemoteDataSource.kt

```diff
@@ -0,0 +1,44 @@
+package com.google.firebase.example.makeitso.data.datasource
+
+import com.google.firebase.auth.EmailAuthProvider
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.example.makeitso.data.model.User
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.tasks.await
+import javax.inject.Inject
+
+class AuthRemoteDataSource @Inject constructor(private val auth: FirebaseAuth) {
+    val currentUser: Flow<User>
+        get() = callbackFlow {
+            val listener =
+                FirebaseAuth.AuthStateListener { auth ->
+                    this.trySend(auth.currentUser?.let {
+                        User(it.uid, isAnonymous = it.isAnonymous)
+                    } ?: User())
```

Comment:
Does sending an empty user (`User()`) actually do anything?
Otherwise we might prefer not to send anything if the user is null (or send that null value).

Reply:
Good point - returning null now.